plaidml2/bridge/keras/backend_test.py

```diff
@@ -476,15 +477,18 @@ def testFlatten(self, b, x):
     def testEye(self, b, *args):
         return b.eye(*args)
 
-    @unittest.skip('TODO: convert to EDSL')
     def testTileIdentity(self):
-        x = pkb.variable(m(3))
-        op = tile.Operation('function (I[N]) -> (O) { O = I; }', [('I', x)],
-                            [('O', tile.Shape(x.shape.dtype, (3,)))])
-        output = op.sole_output().eval()
+        I = pkb.variable(m(3)).tensor
+        N = edsl.TensorDim()
+        #I.bind_dims(N)
+        #I = edsl.TensorOutput(N) #not sure if this is needed ?code segfaults with this line in
+        i = edsl.TensorIndex()
+        I[i] = I[i]
+        opId = pkb._KerasNode('tile_identity', name='opId', tensor=I)
+        pkb.eval(opId)
```

Comment:
I think here, we want:

```
I = pkb.variable(m(3)).tensor
pkb.eval(I)
```

Reply:
Changed as above. Fails with above code in backend_test.py but once moved to edsl_test.py passes with:  

        I = Tensor(LogicalShape(plaidml.DType.FLOAT32, [3]), name='I')
        program = Program('tile_identity', [I])
        self.assertMultiLineEqual(str(program), '''function (
  I[I_0]
) -> (
  _X0
) {
  _X0 = ident(I);
}
''')

is this the expectation?

also should the tests be removed from backend_test.py once they are rewritten in edsl_test.py?

Reply:
That looks right to me. Ideally we'd actually run the program however. That gets a little harder to validate.

@tzerrell did these tests compare to tensorflow in backend_test? I can't imagine they do. Maybe just running them once to ensure they don't crash is enough?

Reply:
They did not compare to TF; they were only checking that they don't crash. My plan was that I'm manually verifying that the code looks correct, but if you could show us how to test running the program here that would be better